internal/gitprovider/gitlab_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package gitprovider
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGitLabProvider_ListBranches(t *testing.T) {
+	tests := []struct {
+		name       string
+		opts       Options
+		want       []Result
+		wantErrMsg string
+	}{
+		{
+			name: "filters branches by regex",
+			opts: Options{
+				Token: os.Getenv("GITLAB_TOKEN"),
```

Comment:
Are we setting this in our GitHub secrets?

Reply:
No this is for local dev, I've been running these tests a lot and with the tokens I'm avoiding rate limits.

Reply:
Oh right. So what's gonna happen in our CI exactly?

Reply:
CI uses anonymous access

Reply:
So we are making 20 API calls per Git provider in every CI run, which shouldn't trigger any rate limits.